src/commonMain/kotlin/dev/romainguy/kotlin/math/Matrix.kt

```diff
@@ -498,6 +502,117 @@ fun rotation(axis: Float3, angle: Float): Mat4 {
     )
 }
 
+/**
+ * Construct a matrix from Euler angles using YPR around ZYX respectively
+ *
+ * The Euler angles are applied in ZYX order.
+ * i.e: a vector is first rotated about X (roll) then Y (pitch) and then Z (yaw).
+ *
+ * @param d Per axis Euler angles in degrees
+ */
+fun euler(d: Float3): Mat4 {
```

Comment:
How is that different from the existing `rotation(Float3)` function?

Reply:
Very good point.
Actually, you are right, there should only be one but they give different results.
I didn't removed the old one because I can't figure out what was its original source equation for this derivation:
https://github.com/romainguy/kotlin-math/blob/191d77967da63dc81580a54fbfb5261343d02b3a/src/commonMain/kotlin/dev/romainguy/kotlin/math/Matrix.kt#L474-L485
To keep the API consistent I just wanted to figure out where did the result rotation matrix come from to tell the matching rotation type it uses.
But, after testing multiple existing converters like [this one](https://www.andre-gaschler.com/rotationconverter/), switching on every angle rotation order possible types, using radians/degrees, normalizing entry values I never get the old rotation matrix result.
I also have the same behavior with the "original" axis angle `rotation(Float3, Float): Mat4`

I tested the results with the introduced `euler(Float3)` and `quaternion(Float4)` and they are always equals to ones from other converters.

Reply:
It's possible there's a bug :) I'd rather you replaced the existing implementation then.

Reply:
Ok I'll delete `rotation(Float3)` and rename `euler(Float3)` to rotation.

>I also have the same behavior with the "original" axis angle rotation(Float3, Float): Mat4

I double checked the axis angle rotation to matrix and the original result is actually good.
This was due to normalized vs not normalized input between the 2 converters tests.

Reply:
I finally realized where does the results differences came from.
Actually it was a row/column inversion because of `Mat4.of()` and `mat4.toFloatArray()` representation.
So the original `rotation(Float3)` was working well.

* Original *
```kotlin
fun rotation(d: Float3): Mat4 {
    val r = transform(d, ::radians)
    val c = transform(r) { x -> cos(x) }
    val s = transform(r) { x -> sin(x) }

    return Mat4.of(
            c.y * c.z, -c.x * s.z + s.x * s.y * c.z, s.x * s.z + c.x * s.y * c.z, 0.0f,
            c.y * s.z, c.x * c.z + s.x * s.y * s.z, -s.x * c.z + c.x * s.y * s.z, 0.0f,
            -s.y, s.x * c.y, c.x * c.y, 0.0f,
            0.0f, 0.0f, 0.0f, 1.0f
    )
```
* New *
```kotlin
fun rotation(d: Float3): Mat4 {
    val r = transform(d, ::radians)
    return eulerZYX(r.z, r.y, r.x)
}
fun eulerZYX(yaw: Float = 0.0f, pitch: Float = 0.0f, roll: Float = 0.0f): Mat4 {
    val cy = cos(yaw)
    val sy = sin(yaw)
    val cp = cos(pitch)
    val sp = sin(pitch)
    val cr = cos(roll)
    val sr = sin(roll)
    return Mat4.of(cy * cp , cy * sp * sr - cr * sy, sp * cr * cy + sr * sy, 0.0f,
            cp * sy, sp * sr * sy + cr * cy, sp * cr * sy - sr * cy, 0.0f,
            -sp, cp * sr, cp * cr, 0.0f,
            0.0f, 0.0f, 0.0f, 1.0f)
}
```
** Should I keep the functions version more comprehension on what Euler rotation type is used ? **

Reply:
No, just add a short doc to the existing rotation function explaining that the order of rotations is ZYX.

Reply:
I didn't resist on adding a rotations order parameter: `rotation(Float3, RotationOrder) : Mat4`
So the lib can now handled every major possible rotations.
Euler angle rotation Matrix uses the most common ZYX by default.
But other orders are also handled.

Reply:
And everyone has been tested

Reply:
Fixed https://github.com/romainguy/kotlin-math/pull/33/commits/e60495ba84a929cbda6ede258a067ad5baa72d13